spacy_curated_transformers/cli/fill_config_transformer.py

```diff
@@ -0,0 +1,265 @@
+            if value is None:
+                msg.fail(
+                    f"Hugging Face tokenizer config has a missing key '{hf_key}'",
+                    exits=1,
+                )
+        filled_params[param_name] = value
+
+    msg.info(title="Filled-in model parameters")
+    msg.table(filled_params)
+
+    trf_config = config["components"][transformer_name]["model"]
+    trf_config.update(filled_params)
+
+    is_stdout = str(output_path) == "-"
+    save_config(config, output_path, is_stdout=is_stdout, silent=is_stdout)
+
+
+def _get_hf_model_config(name: str, revision: str) -> Dict[str, Any]:
+    from huggingface_hub import hf_hub_download
+
+    config_path = hf_hub_download(
+        repo_id=name, filename="config.json", revision=revision
+    )
+    return _parse_json_file(config_path)
+
+
+def _parse_json_file(path: str) -> Dict[str, Any]:
+    with open(path, encoding="utf-8") as f:
+        return json.load(f)
+
+
+def _get_encoder_loader_config(
+    config: Dict[str, Any], transformer_name: str
+) -> Optional[Dict[str, Any]]:
+    try:
+        loader = config["initialize"]["components"][transformer_name]["encoder_loader"]
+        assert "HFTransformerEncoderLoader" in loader["@model_loaders"]
+        return loader
+    except (KeyError, AssertionError):
+        return None
+
+
+def _get_curated_transformer_arch(
+    config: Dict[str, Any], transformer_name: str
+) -> Optional[str]:
+    try:
+        transformer = config["components"][transformer_name]["model"]
+        trf_arch_splits = transformer["@architectures"].split(".")
+        assert len(trf_arch_splits) == 3 and trf_arch_splits[-2].endswith("Transformer")
+        return trf_arch_splits[-2]
+    except (KeyError, AssertionError):
+        return None
+
+
+def _load_hf_tokenizer(model_name: str, model_revision: str, msg: Printer) -> Any:
+    # The entry point params that we need to fetch from the upstream HF tokenizer
+    # are not consistently serialized across all available models. So, we'll just
+    # load the tokenizer using the `transformers` library and directly read the params
+    # from the instance.
+    from transformers import AutoTokenizer
+
+    try:
+        hf_tokenzier = AutoTokenizer.from_pretrained(
+            model_name, revision=model_revision
+        )
+
+        # The `model_max_length` value can sometimes be set to a very large integer
+        # if the model doesn't have an upperbound on its input length. Saving this
+        # as-is to the config will raise an error during serialization. So, we'll
+        # try to look up the correct value manually. If we don't find one, fallback
+        # to a smaller (large) value that can be serialized correctly.
+        # cf: https://github.com/huggingface/transformers/blob/224da5df6956340a5c680e5b57b4914d4d7298b6/src/transformers/tokenization_utils_base.py#L104
+        if hf_tokenzier.model_max_length == int(1e30):
+            hf_tokenzier.model_max_length = 9999
```

Comment:
This seems potentially too low? Maybe int32.max so it's a more obvious max?

Reply:
Done.